modules/renter/persist.go

```diff
@@ -112,15 +111,49 @@ func (r *Renter) ShareFiles(nicknames []string, sharedest string) error {
 	for _, nickname := range nicknames {
 		file, exists := r.files[nickname]
 		if !exists {
-			return ErrUnknownNickname
+			return nil, ErrUnknownNickname
 		}
 		rsf.Files = append(rsf.Files, *file)
 	}
 
 	shareBytes, err := json.Marshal(rsf)
 	if err != nil {
+		return nil, err
+	}
+
+	// Gzip the result.
+	var zipBuffer bytes.Buffer
+	zip, err := gzip.NewWriterLevel(&zipBuffer, flate.BestCompression)
+	if err != nil {
+		return nil, err
+	}
+	_, err = zip.Write(shareBytes)
+	if err != nil {
+		return nil, err
+	}
+	zip.Close()
+
+	return zipBuffer.Bytes(), nil
```

Comment:
since you wind up writing the zip data to a file, you should just pass the file to `shareBytes` as an `io.Writer` and create the `gzip.NewWriterLevel` with that.